### PR TITLE
add sleep after new block to the request_decider

### PR DESCRIPTION
**Pull request by Jiloc**
## Description

Part of : #1237

## Changes
 - originally part of #1243 , this change add a new `requests_processing_delay` to the request_decider. This will give additional time to the bitcoin nodes to get in sync before receiving the vote decisions from the signers. 
 
## Checklist:

- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by djordon — APPROVED** [review]
Looks good.

Let's wait to get more eyes on this before merging, in case there is some edge case that we're blanking on that would make this a bad idea.

**Review by matteojug — COMMENTED** [review]
Thinking out loud: assuming everyone gets the bitcoin block at exactly the same time, with the sleep as is here we may risk having the coordinator starting the round before the decisions manage to get to it (like, everyone broadcast its own decision, then the coordinator starts but it only has its own decision).
I think this is slightly worse than before, but it could be equivalent.

**Comment by Jiloc** [comment]
How about we separate the two timeouts: one in the RequestDecider after a new block, and another in the TxCoordinator after receiving the signal from the RequestDecider? We could set them both to 15 seconds.

While this isn't an ideal or final solution, I believe it's a reasonable compromise. It should still allow us to receive the decisions and blocks in time most of the time.

#1243  will remain the final solution. And if we implement a filter to only resend messages that didn't receive all replies, we could significantly reduce the number of repeated messages.

File: signer/src/config/mod.rs [hunk 1]
```diff
@@ -223,9 +226,14 @@ pub struct SignerConfig {
     pub bootstrap_signatures_required: u16,
     /// The number of seconds the coordinator will wait
     /// before processing a new Bitcoin block
-    /// (allowing it to propagate to the others signers)
+    /// (allowing it to propagate the request decisions the others signers)
```
Comment (matteojug) [inline signer/src/config/mod.rs:229]: very nit:
```suggestion
    /// (allowing the request decisions to propagate to the others signers)
```

File: signer/src/config/mod.rs [hunk 2]
```diff
@@ -223,9 +226,14 @@ pub struct SignerConfig {
...
     #[serde(deserialize_with = "duration_seconds_deserializer")]
     pub bitcoin_processing_delay: std::time::Duration,
+    /// The number of seconds the request decider will wait
+    /// before processing the new sbtc requests
+    /// (allowing it to propagate the bitcoin block the others signers)
```
Comment (matteojug) [inline signer/src/config/mod.rs:234]: very nit:
```suggestion
    /// (allowing the bitcoin block to propagate to the others signers)
```

File: signer/src/config/mod.rs [hunk 3]
```diff
@@ -223,9 +226,14 @@ pub struct SignerConfig {
...
+    #[serde(deserialize_with = "duration_seconds_deserializer")]
+    pub requests_processing_delay: std::time::Duration,
```
Comment (matteojug) [inline signer/src/config/mod.rs:236]: Do we want this to be a required param? (probably yes, just to confirm)
If so, can you add the `breaking-local` tag to the PR?

**Review by matteojug — APPROVED** [review]

File: signer/src/config/mod.rs (hunk 3 above)
Comment (Jiloc) [reply signer/src/config/mod.rs:236]: Alternatively, we could give a default value?

Comment (matteojug) [reply signer/src/config/mod.rs:236]: It makes sense to have this required (as we did for the other sleep), the only nuisance is that we don't know what's the best value for this on mainnet so we may require signers to update the config more than once. But even with defaults we would still need them to update the config (or do a new release, that is even worse). Imho it's okay to have it required.
